tests/phpunit/api/v4/Action/BasicCustomFieldTest.php

```diff
@@ -277,4 +277,39 @@ public function testRelationshipCacheCustomFields() {
     $this->assertEquals('Buddy', $results[0]["$cgName.PetName"]);
   }
 
+  /**
+   * Some types are creating a dummy option group even if we don't have
+   * any option values.
+   * @throws \API_Exception
+   */
+  public function testUndesiredOptionGroupCreation(): void {
```

Comment:
Note this 'undesired' creation still happens in v3 - because the tests tell us it is 'by design' in v3. However, I think it's ok to have a different opinion on this in v4

Reply:
Thanks @eileenmcnaughton will take a look!

Reply:
Interesting. Yeah it seems it did used to create an extra option group in v3 even if you didn't specify some values, if e.g. you didn't specify option_group_id (e.g. in 5.37 - https://github.com/civicrm/civicrm-core/blob/5.37.0/tests/phpunit/CiviTest/CiviUnitTestCase.php#L1621 - the "default_value" doesn't make a difference in case anyone's wondering). As noted in #20743 api3 used to set option_type=1 by default via spec, which would then trigger the if statement if it was a certain data type etc.